extensions/blocks/map/map.php

```diff
@@ -25,7 +25,85 @@
 function jetpack_map_block_load_assets( $attr, $content ) {
 	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		static $map_block_counter = [];
+
+		$id = get_the_ID();
+		if ( ! $map_block_counter[ $id ] ) {
+			$map_block_counter[ $id ] = 0;
+		}
+		$map_block_counter[ $id ]++;
+
+		$iframe_url = add_query_arg(
+			array(
+				'map-block-counter' => $map_block_counter[ get_the_ID() ],
+				'map-block-post-id' => get_the_ID(),
+			),
+			get_permalink()
+		);
+
+		$placeholder = preg_replace( '/(?<=<div\s)/', 'placeholder ', $content );
+
+		return sprintf(
+			'<amp-iframe src="%s" width="%d" height="%d" layout="responsive" allowfullscreen sandbox="allow-scripts">%s</amp-iframe>',
+			esc_url( $iframe_url ),
+			absint( Jetpack::get_content_width() ),
+			absint( Jetpack::get_content_width() * 0.75 ),
+			$placeholder
+		);
+	}
+
 	Jetpack_Gutenberg::load_assets_as_required( 'map' );
 
 	return preg_replace( '/<div /', '<div data-api-key="' . esc_attr( $api_key ) . '" ', $content, 1 );
 }
+
+/**
+ * Render a page containing only a single Map block.
+ */
+function jetpack_map_block_render_single_block_page() {
+	$map_block_counter = (int) filter_input( INPUT_GET, 'map-block-counter', FILTER_SANITIZE_NUMBER_INT );
+	$map_block_post_id = (int) filter_input( INPUT_GET, 'map-block-post-id', FILTER_SANITIZE_NUMBER_INT );
+	if ( ! $map_block_counter || ! $map_block_post_id ) {
+		return;
+	}
+
+	/* Create an array of all root-level DIVs that are Map Blocks */
+	$post = get_post( $map_block_post_id );
+
+	$post_html = new DOMDocument();
+	$post_html->loadHTML( $post->post_content );
+	$xpath          = new DOMXPath( $post_html );
+	$map_block_divs = $xpath->query( '//div[ contains( @class, "wp-block-jetpack-map" ) ]' );
```

Comment:
```suggestion
	$map_block_div = $xpath->query( '//div[ contains( @class, "wp-block-jetpack-map" ) ]' )->item( $map_block_counter - 1 );
```

Then below you can just do:

```php
if ( ! $map_block_div ) {
    return;
}
```

And refer to that instead of getting the index of an array.

Reply:
Nice https://github.com/Automattic/jetpack/pull/13405/commits/e9d604d8d2f8caace8b178654651373a93a25536